platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt

```diff
@@ -79,6 +96,33 @@ abstract class GodotActivity : FragmentActivity(), GodotHost {
 		}
 	}
 
+
+	@CallSuper
+	protected open fun getNewGodotInstanceIntent(args: Array<String>): Intent {
+		return Intent().setComponent(ComponentName(this, javaClass.name))
+			.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+			.putExtra(EXTRA_COMMAND_LINE_PARAMS, args)
+	}
+
+	override fun onNewGodotInstanceRequested(args: Array<String>): Int {
+		Log.d(TAG, "Restarting with parameters ${args.contentToString()}")
+		triggerRebirth(null, getNewGodotInstanceIntent(args))
+		// fake 'process' id returned by create_instance() etc
+		return DEFAULT_WINDOW_ID;
+	}
```

Comment:
Adding the `triggerRebirth` function is a good idea! However, I don't see similar benefits with `getNewGodotInstanceIntent` and `onNewGodotInstanceRequested`. The latter especially is different enough between the two classes that there's not much benefit in trying to extract common functionality, so I'd just leave it as is.

Reply:
This functionality is needed here since I removed it from GodotApp to avoid kotlin/java interop..

The default behavior (here) is used by GodotApp, BaseGodotEditor doesn't call `super.onNewGodotInstanceRequested(...)` since the only common functionality is `triggerRebirth()`

I feel that addFlags() and putExtra() _are_ enough common functionality between GodotActivity and BaseGodotEditor, since they both need those basic intent options configured to function. (BaseGodotEditor just calls `intent.setComponent()` again to override the classname.

Reply:
> This functionality is needed here since I removed it from GodotApp to avoid kotlin/java interop..

What's the interop issue you're referring to?

> For getNewGodotInstanceIntent():
I feel that addFlags() and putExtra() are enough common functionality between GodotActivity and BaseGodotEditor, since they both need those basic intent options configured to function. (BaseGodotEditor just calls intent.setComponent() again to override the classname.

Let's remove this one.. creating an intent is simple enough that we don't really need an helper method for it. In addition, each subclass has different requirements on how they need to configure their intent and the arguments they're passing to it.

Reply:
>>    This functionality is needed here since I removed it from GodotApp to avoid kotlin/java interop..

> What's the interop issue you're referring to?

I interop _works_ fine but the impedance mismatch is [higher than expected](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) to call across the two. [Specifically this
`getEXTRA_COMMAND_LINE_PARAMS()` quirk](https://github.com/godotengine/godot/pull/101050#discussion_r1901462819) caused me to decide that It's better to just write as much of this change in `GodotActivity.kt` to avoid crossing between java and kotlin.


> Let's remove this one..

Done.